packages/cozy-client/src/store/documents.js

```diff
@@ -107,3 +92,61 @@ export const getDocumentFromSlice = (state = {}, doctype, id) => {
   }
   return state[doctype][id]
 }
+
+const extractAndMergeDocument = (data, updatedStateWithIncluded) => {
+  const doctype = data[0]._type
+
+  if (!doctype) {
+    throw new Error('Document without _type', data[0])
+  }
+  const ordonnedData = keyBy(data, properId)
+
+  let mergedData = {}
+  if (updatedStateWithIncluded && updatedStateWithIncluded[doctype]) {
+    Object.values(updatedStateWithIncluded[doctype]).map(dataState => {
+      if (!mergedData[doctype]) mergedData[doctype] = {}
+      if (ordonnedData[dataState._id]) {
+        mergedData[doctype] = {
+          ...mergedData[doctype],
+          [dataState._id]: {
+            ...dataState,
+            ...ordonnedData[dataState._id],
+            ...mergedData[doctype][dataState._id]
+          }
+        }
+      } else {
+        mergedData[doctype] = {
+          ...mergedData[doctype],
+          [dataState._id]: {
+            ...dataState,
+            ...mergedData[doctype][dataState._id]
+          }
+        }
+      }
+    })
+  }
+  Object.values(ordonnedData).map(data => {
+    if (!mergedData[doctype]) mergedData[doctype] = {}
+    if (mergedData[doctype][data._id]) {
+      mergedData[doctype] = {
+        ...mergedData[doctype],
+        [data._id]: {
+          ...mergedData[doctype][data._id],
+          ...data
+        }
+      }
+    } else {
+      mergedData[doctype] = {
+        ...mergedData[doctype],
+        [data._id]: {
+          ...data
+        }
+      }
+    }
+  })
+
+  return {
+    ...updatedStateWithIncluded,
```

Comment:
Do we need to return `updatedStateWithIncluded ` here? If I understood correctly, all the data is already in `mergedData`.